src/github.com/getlantern/peerscanner/group.go

```diff
@@ -39,11 +39,11 @@ func (g *group) deregister(h *host) {
 
 	log.Debugf("Deregistering from %v: %v", g.subdomain, h)
 
+	// Destroy the record in the rotation...
+	err := cfutil.DestroyRecord(g.existing)
```

Comment:
Opportunistic fix.  Previously, destroying the g.existing record never worked because we had already cleared it.